pkg/operator/target_config_reconciler.go

```diff
@@ -362,14 +447,37 @@ func isResourceRegistered(discoveryClient discovery.DiscoveryInterface, gvk sche
 	return false, nil
 }
 
+func setDNSNamesToUnstructured(certAsUnstructured *unstructured.Unstructured, varPrefix, serviceName, serviceNamespace string) error {
+	dnsNames, found, err := unstructured.NestedStringSlice(certAsUnstructured.Object, "spec", "dnsNames")
+	if !found || err != nil {
+		return fmt.Errorf("%v: .spec.dnsNames not found: %v", certAsUnstructured.GetName(), err)
+	}
+	for i := range dnsNames {
+		dnsNames[i] = injectService(varPrefix, dnsNames[i], serviceName, serviceNamespace)
+	}
+	return unstructured.SetNestedStringSlice(certAsUnstructured.Object, dnsNames, "spec", "dnsNames")
+}
+
+func injectService(varPrefix, value, serviceName, serviceNamespace string) string {
+	switch {
+	case strings.Index(value, "$(") >= 0:
+		value = strings.Replace(value, fmt.Sprintf("$(%s_NAME)", varPrefix), serviceName, 1)
+		value = strings.Replace(value, fmt.Sprintf("$(%s_NAMESPACE)", varPrefix), serviceNamespace, 1)
+	case strings.Index(value, "${") >= 0:
```

Comment:
I couldn't find any resource with the prefix `${`. Is there any reason or just to be on the safe side?. I think, if there is any format changes in the future, our CI will detect this. Maybe it is better to just replace according to `$(`, WDYT?.

Reply:
It is used in this patch https://github.com/kubernetes-sigs/jobset/blob/ccd5c4e7cee7dd312052b35fa7196c1722f96071/config/components/prometheus/monitor_tls_patch.yaml#L23-L34 as it was not picked by kustomize and had to be replaced later.

Reply:
Makes sense. Thank you.